### PR TITLE
Mech ability/utility modules

### PR DESCRIPTION
**Pull request by TiviPlus**
Adds for starters:
Zoom
actuator overload
cloak smoke
tanglefoot smoke

### CODE REVIEW

File: code/modules/vehicles/mecha/equipment/tools/greyscale_tools.dm [hunk 1]
```diff
@@ -27,3 +27,78 @@
...
+/obj/item/mecha_parts/mecha_equipment/ability/smoke/tanglefoot
+	name = "tanglefoot generator"
+	desc = "A tanglefoot smoke generator capable of dispensing large amounts of non-lethal gas that saps the energy from any xenoform creatures it touches."
+	mech_flags = EXOSUIT_MODULE_GREYSCALE
+	ability_to_grant = /datum/action/vehicle/sealed/mecha/mech_smoke
+	smoke_type = /datum/effect_system/smoke_spread/plasmaloss
+
+/obj/item/mecha_parts/mecha_equipment/ability/smoke/cloak_smoke
+	name = "smoke generator"
+	desc = "generates a large amount of cloaking smoke to disguise nearby friendlies. Sadly, huge robots are too difficult to hide with it."
```
Comment (XSlayer300) [inline code/modules/vehicles/mecha/equipment/tools/greyscale_tools.dm:101]:
```suggestion
	desc = "A multiple launch module that generates a large amount of cloaking smoke to disguise nearby friendlies. Sadly, huge robots are too difficult to hide with it."
```

**Review by Kuro020 — APPROVED** [review]

File: code/modules/vehicles/mecha/equipment/tools/greyscale_tools.dm [hunk 2]
```diff
@@ -62,3 +62,98 @@
...
 	return ..()
+
+
+
+/obj/item/mecha_parts/mecha_equipment/ability
+	name = "generic mech ability"
+	desc = "You shouldnt be seeing this"
+	equipment_slot = MECHA_UTILITY
+	///if given, a single flag of who we want this ability to be granted to
+	var/flag_controller
```
Comment (BraveMole) [inline code/modules/vehicles/mecha/equipment/tools/greyscale_tools.dm:73]: = NONE

File: code/modules/vehicles/mecha/equipment/tools/greyscale_tools.dm [hunk 3]
```diff
@@ -62,3 +62,98 @@
...
+	///typepath of ability we want to grant
+	var/ability_to_grant
+	///reference to image that is used as an overlay
+	var/image/overlay
+
+/obj/item/mecha_parts/mecha_equipment/ability/Initialize()
+	. = ..()
+	if(icon_state)
```
Comment (BraveMole) [inline code/modules/vehicles/mecha/equipment/tools/greyscale_tools.dm:81]: Call update icon imo

File: code/modules/vehicles/mecha/equipment/tools/greyscale_tools.dm [hunk 4]
```diff
@@ -62,3 +62,98 @@
...
+		overlay = image('icons/mecha/mecha_ability_overlays.dmi', icon_state = icon_state, layer = 10)
+
+/obj/item/mecha_parts/mecha_equipment/ability/attach(obj/vehicle/sealed/mecha/M, attach_right)
+	. = ..()
+	M.add_overlay(overlay)
```
Comment (BraveMole) [inline code/modules/vehicles/mecha/equipment/tools/greyscale_tools.dm:86]: Ditto

File: code/modules/vehicles/mecha/equipment/tools/greyscale_tools.dm [hunk 5]
```diff
@@ -62,3 +62,98 @@
...
+	if(flag_controller)
+		M.initialize_controller_action_type(ability_to_grant, flag_controller)
+	else
+		M.initialize_passenger_action_type(ability_to_grant)
+
+/obj/item/mecha_parts/mecha_equipment/ability/detach(atom/moveto)
+	chassis.cut_overlay(overlay)
```
Comment (BraveMole) [inline code/modules/vehicles/mecha/equipment/tools/greyscale_tools.dm:93]: Ditto

File: code/modules/vehicles/mecha/combat/gygax.dm [hunk 6]
```diff
@@ -37,6 +37,18 @@
 		chassis.leg_overload_mode = !chassis.leg_overload_mode
 	action_icon_state = "mech_overload_[chassis.leg_overload_mode ? "on" : "off"]"
 	chassis.log_message("Toggled leg actuators overload.", LOG_MECHA)
+	//tgmc add
+	var/obj/item/mecha_parts/mecha_equipment/ability/dash/ability = locate() in chassis.equip_by_category[MECHA_UTILITY]
+	if(ability)
+		chassis.cut_overlay(ability.overlay)
+		var/state = chassis.leg_overload_mode ? (initial(ability.icon_state) + "_active") : initial(ability.icon_state)
+		ability.overlay = image('icons/mecha/mecha_ability_overlays.dmi', icon_state = state, layer=10)
+		chassis.add_overlay(ability.overlay)
```
Comment (BraveMole) [inline code/modules/vehicles/mecha/combat/gygax.dm:46]: Update icon ? This time less convinced

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/vehicles/mecha/equipment/tools/greyscale_tools.dm (hunk 2 above)
Comment (TiviPlus) [reply code/modules/vehicles/mecha/equipment/tools/greyscale_tools.dm:73]:
```suggestion
	var/flag_controller = NONE
```

**Pull request merged**